src/Chainweb/Pact/Backend/ChainwebPactDb.hs

```diff
@@ -120,25 +120,36 @@ doReadRow
     -> BlockHandler SQLiteEnv (Maybe v)
 doReadRow d k = forModuleNameFix $ \mnFix ->
     case d of
-        KeySets -> lookupWithKey (convKeySetName k)
+        KeySets -> lookupWithKey (convKeySetName k) noCache
         -- TODO: This is incomplete (the modules case), due to namespace
         -- resolution concerns
-        Modules -> lookupWithKey (convModuleName mnFix k)
-        Namespaces -> lookupWithKey (convNamespaceName k)
-        (UserTables _) -> lookupWithKey (convRowKey k)
-        Pacts -> lookupWithKey (convPactId k)
+        Modules -> lookupWithKey (convModuleName mnFix k) checkModuleCache
+        Namespaces -> lookupWithKey (convNamespaceName k) noCache
+        (UserTables _) -> lookupWithKey (convRowKey k) noCache
+        Pacts -> lookupWithKey (convPactId k) noCache
   where
     tableName = domainTableName d
     (Utf8 tableNameBS) = tableName
 
+    checkModuleCache u b = MaybeT $ do
+        txid <- use bsTxId -- cache priority
+        mc <- use bsModuleCache
+        (r, mc') <- liftIO $ checkDbCache u b txid mc
+        modify' (bsModuleCache .~ mc')
```

Comment:
I think lines 136-138 could be simply `modify' $ liftIO . checkDbCache u b txid`?

Reply:
Also this was originally down by `lookupInDb` and `noCache`, is there a reason it's up here so far away from those?

Reply:
> I think lines 136-138 could be simply modify' $ liftIO . checkDbCache u b txid?

Unfortunately `modify'` returns just `()`. There is probably a way to do it by combining `MaybeT`, and `StateT` with some lensy tricks, but that's a bit above my production type proficiency.